### DISCUSSION TITLE
Welcome to FinRL-Meta Discussions!

### DISCUSSION THREAD

**Discussion by YangletLiu** [AI4Finance-Foundation/FinRL-Meta#274] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by abdulazeezb** [comment]
**Feeling Thankful!!!**
Hello Xiao-Yang Liu, Hope you are doing Great
First of All, Thank you so much for creating/contributing to FinRL, It would be great repo for all the stock market enthusiastic 

**Intro About Me**

I'm working as Data Scientist and as my personal project working on predicting the stock market.

**Question**
I'm training a model for Day Trading(all the stocks will close in EOD), How are we sharing this information to agent(to sell all the stocks at the end of the day) in StockTrading Environment "finrl/meta/env_stock_trading/env_stocktrading.py"

**Reply by YangletLiu** [reply]
> **Feeling Thankful!!!** Hello Xiao-Yang Liu, Hope you are doing Great First of All, Thank you so much for creating/contributing to FinRL, It would be great repo for all the stock market enthusiastic
> 
> **Intro About Me**
> 
> I'm working as Data Scientist and as my personal project working on predicting the stock market.
> 
> **Question** I'm training a model for Day Trading(all the stocks will close in EOD), How are we sharing this information to agent(to sell all the stocks at the end of the day) in StockTrading Environment "finrl/meta/env_stock_trading/env_stocktrading.py"

Let us assume you are working on a minute level, about 390 minutes each trading day.
1). Each minute, your state can consist of [balance_t, shares_t, price_t, feature_t]
2). For feature_t (or technical indicators at t), you can use any feature/indicators calculated from a past window of size T; either use traditional indicators, say TA-lib; or use LSTM/Transforms to obtain learned features;
3). Now, in your reward calculation, since you want to sell all stock at EOD,
      how about adding a punishment term in the window [EOD - M, EOD]?  Try to choose a decaying function for your purpose.
